baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/BracesRequired.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.DoWhileLoopTree;
+import com.sun.source.tree.EnhancedForLoopTree;
+import com.sun.source.tree.ForLoopTree;
+import com.sun.source.tree.IfTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.WhileLoopTree;
+import javax.annotation.Nullable;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "BracesRequired",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = BugPattern.SeverityLevel.WARNING,
+        summary = "Braces are required for readability")
+public final class BracesRequired extends BugChecker implements
+        BugChecker.DoWhileLoopTreeMatcher,
+        BugChecker.ForLoopTreeMatcher,
+        BugChecker.EnhancedForLoopTreeMatcher,
+        BugChecker.IfTreeMatcher,
+        BugChecker.WhileLoopTreeMatcher {
+
+    @Override
+    public Description matchIf(IfTree tree, VisitorState state) {
+        check(tree.getThenStatement(), state);
+        check(tree.getElseStatement(), state);
+        return Description.NO_MATCH;
```

Comment:
seems a little odd to always return NO_MATCH and rely on a side effect of check to actually do the fix

Reply:
Agreed, it's a bit odd. It allows us to keep the code cleaner and reusable without creating potentially unnecessary SuggestedFix.Builder objects for every conditional and loop. Several of the upstream checks are structured similarly.